copy/copy.go

```diff
@@ -646,6 +689,26 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 	// If encrypted and decryption keys provided, we should try to decrypt
 	ic.diffIDsAreNeeded = ic.diffIDsAreNeeded || (isEncrypted(src) && ic.c.ociDecryptConfig != nil) || ic.c.ociEncryptConfig != nil
 
+	// If enabled, fetch and compare the destination's manifest. And as an optimization skip updating the destination iff equal
+	if options.OptimizeDestinationImageAlreadyExists {
+		shouldUpdateSigs := len(sigs) > 0 || options.SignBy != "" // TODO: Consider allowing signatures updates only and skipping the image's layers/manifest copy if possible
+		noPendingManifestUpdates := ic.noPendingManifestUpdates()
+
+		logrus.Debugf("Checking if we can avoid the copy: has signatures=%t, OCI encryption=%t, no manifest updates=%t", shouldUpdateSigs, destRequiresOciEncryption, noPendingManifestUpdates)
+		if !shouldUpdateSigs && !destRequiresOciEncryption && noPendingManifestUpdates {
```

Comment:
Can you add a `// TODO: Consider compression changes` here?

Just in case we want further optimizations in the future.

Reply:
IIRC (a drive-by comment without reading much else), requested compression changes are currently ignored anyway if `TryReusingBlob` finds a match — and if we find the full manifest, that should always be the case. So requested compression changes should not be a reason to skip this optimization.

Reply:
Thanks for the explanation @mtrmac

To paraphrase/If I understand correctly: further work would be needed for change in compression to be effected in a copy? Only then, would it make sense to disable the optimization (or more specifically, then it would be required to disable the optimization).

I've added the `// TODO: ..` as per @vrothberg's suggestion. But I don't see anything else required here(?)

Reply:
No, I’m saying that if the destination manifest exists, any requested re-compression via `types.SystemContext.Compression{Operation,Algorithm}` wouldn’t happen _anyway_, so there is nothing to be done here.

OTOH I’m not 100% sure what “consider compression changes” refers to, @vrothberg can you clarify it into something more precise / actionable?

Reply:
I am cool to drop the comment if it's causing more confusion than good.  My intention was to leave some breadcrumbs in case `skopeo sync` would support compression changes.  As far as I understand the code, compression changes would be ignored and @mtrmac's comment above confirm.

Reply:
I've remove the comment. Thanks